cli/pkg/kctrl/cmd/app/init/init.go

```diff
@@ -0,0 +1,230 @@
+	isAppCommandRunExplicitly bool
+}
+
+func NewCreateStep(ui cmdcore.AuthoringUI, build Build, logger logger.Logger, depsFactory cmdcore.DepsFactory, isAppCommandRunExplicitly bool) *CreateStep {
+	return &CreateStep{
+		ui:                        ui,
+		build:                     build,
+		logger:                    logger,
+		depsFactory:               depsFactory,
+		isAppCommandRunExplicitly: isAppCommandRunExplicitly,
+	}
+}
+
+func (createStep CreateStep) GetAppBuild() Build {
+	return createStep.build
+}
+
+func (createStep *CreateStep) PreInteract() error {
+	return nil
+}
+
+func (createStep *CreateStep) Interact() error {
+	fetchConfiguration := NewFetchStep(createStep.ui, createStep.build, createStep.isAppCommandRunExplicitly)
+	err := Run(fetchConfiguration)
+	if err != nil {
+		return err
+	}
+
+	templateConfiguration := NewTemplateStep(createStep.ui, createStep.build)
+	err = Run(templateConfiguration)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (createStep *CreateStep) PostInteract() error {
+	appSpec := createStep.build.GetAppSpec()
+	if appSpec.Deploy == nil {
+		appSpec.Deploy = []kcv1alpha1.AppDeploy{kcv1alpha1.AppDeploy{Kapp: &kcv1alpha1.AppDeployKapp{}}}
+	}
+	createStep.build.SetAppSpec(appSpec)
+	createStep.configureExportSection()
+
+	if createStep.isAppCommandRunExplicitly {
+		appConfig, err := createStep.generateApp()
+		if err != nil {
+			return err
+		}
+
+		appContent, err := yaml.Marshal(appConfig)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(AppFileName, appContent)
+		if err != nil {
+			return err
+		}
+
+		createStep.ui.PrintInformationalText(fmt.Sprintf("Both the files can be accessed from the following location:"))
+	}
+
+	return nil
+}
+
+func (createStep CreateStep) generateApp() (kcv1alpha1.App, error) {
+	var app kcv1alpha1.App
+	exists, err := IsFileExists(AppFileName)
+	if err != nil {
+		return kcv1alpha1.App{}, err
+	}
+
+	if exists {
+		data, err := os.ReadFile(AppFileName)
+		if err != nil {
+			return kcv1alpha1.App{}, err
+		}
+		err = yaml.Unmarshal(data, &app)
+		if err != nil {
+			return kcv1alpha1.App{}, err
+		}
+	} else {
+		app = createStep.createAppFromAppBuild()
+	}
+	timestamp := v1.NewTime(time.Now().UTC()).Rfc3339Copy()
+	app.ObjectMeta.CreationTimestamp = timestamp
+	return app, nil
+
+}
+
+func printNextStep(ui cmdcore.AuthoringUI) {
+	ui.PrintInformationalText("\n**Next steps**")
+	ui.PrintInformationalText("\nCreated app can be consumed in following ways:\n")
+}
+
+func printInformation(ui cmdcore.AuthoringUI) {
+	ui.PrintInformationalText("\n**Information**\napp-build.yml is generated as part of this flow. This file can be used for further updating and adding complex scenarios while using the `kctrl dev deploy` command. Please read the link'ed documentation for more explanation.")
+}
+
+func (createStep CreateStep) createAppFromAppBuild() kcv1alpha1.App {
+	//TODO Should we ask for the app Name ?
+	appName := "microservices-demo"
```

Comment:
we should ask for the name, default: directory name.

Reply:
Adopted